guides/14_Content_Security_Policy.md

```diff
@@ -0,0 +1,38 @@
+# Content Security Policy for applications using the Amazon Chime SDK
+
+Modern web applications use [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) to protect users from certain classes of attacks. You can add a security policy by configuring your web server to return the Content-Security-Policy HTTP header or using <meta> element to configure a policy. A guide to use a Lambda@Edge function to inject security headers through CloudFront, see [Adding HTTP Security Headers Using Lambda@Edge and Amazon CloudFront](https://aws.amazon.com/blogs/networking-and-content-delivery/adding-http-security-headers-using-lambdaedge-and-amazon-cloudfront/).
+
+## Meetings connectivity
+
+All applications that use voice and video connectivity — _e.g._, `AudioVideoFacade` classes — will need:
+
+`connect-src`: `*.chime.aws` to reach signaling servers.
+
+## Messaging
+
+* Cognito: 
+
+You can use Amazon Cognito for user authentication when using the Amazon Chime SDK messaging feature. For an example of how to use Amazon Cognito with the Amazon Chime SDK messaging features, see [Build chat features into your application with Amazon Chime SDK messaging](https://aws.amazon.com/blogs/business-productivity/build-chat-features-into-your-application-with-amazon-chime-sdk-messaging/). 
+
+You can inlcude the Amazon AWS endpoint in your policy to allow the Amazon Chime SDK access to the Amazon Cognito for security purpose:
+`connect-src`: `*.amazonaws.com`.
+
+## Logging
+
+The Amazon Chime SDK for JavaScript listens to securitypolicyviolation events and logs blocked URI during signaling connection for join meetings. An error code starting with "Security Policy Violation" will be printed in the console log if there is securitypolicyviolation event happens. Since this is an [experimental technology](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Conventions_definitions#experimental), some certain browsers might not support this security policy violation event detection. Check the [Browser compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent#browser_compatibility) carefully before using this in production.
+
+## Amazon Voice Focus
+
+As discussed in the [Amazon Voice Focus guide](amazonvoice_focus.html), you will need to include the following in your policy to allow the Amazon Chime SDK access to resources it needs at runtime to activate Amazon Voice Focus:
```

Comment:
- The `Amazon Voice Focus guide` link is broken. You need to specify the full path to link the generated `html` file.

- How about this?

> To use the [Amazon Voice Focus](https://aws.github.io/amazon-chime-sdk-js/modules/amazonvoice_focus.html), ensure that your policy includes the following:
>
> * `script-src` and ...
> * ...
>
> See the [Amazon Voice Focus CSP section](https://aws.github.io/amazon-chime-sdk-js/modules/amazonvoice_focus.html#content-security-policy) for more information.

Reply:
Will fix. Thanks!